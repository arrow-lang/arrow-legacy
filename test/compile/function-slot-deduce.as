extern def exit(a: int);

def method() {
  let a;
  a = 20;

  exit(a);
}

method();
