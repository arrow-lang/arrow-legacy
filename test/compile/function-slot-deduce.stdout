define internal void @"function-slot-deduce.@init"() {
  call void @method()
  ret void
}

define internal void @method() {
  %a = alloca i128
  store i128 20, i128* %a
  %1 = load i128* %a
  call void @exit(i128 %1)
  ret void
}

declare void @exit(i128 signext)

define i32 @main(i32, i8**, i8**) {
  call void @"function-slot-deduce.@init"()
  ret i32 0
}
